django/db/models/fields/related_descriptors.py

```diff
@@ -798,6 +798,81 @@ def _get_set_deprecation_msg_params(self):
             self.rel.get_accessor_name() if self.reverse else self.field.name,
         )
 
+    def add_relations(self, relations, *, assert_no_collisions=False):
+        """
+        Creates many (M1, M2) relations with O(1) database queries.
+
+        Any requested relations that already exist will be left alone.
+
+        If you assert that none of the requested relations already exist,
+        you can pass assert_no_collisions=True to save 1 database query.
+        """
+        if len(relations) == 0:
+            return
+
+        (M1_M2, m1_pk_field_name, m2_pk_field_name, m1_m2_pk_tuples) = \
```

Comment:
The parenthesis here are unneeded

Reply:
Can we also rename `M1_M2` to `through_model` or somesuch?

Reply:
I like providing parens when working with tuples. Especially long ones like this. Does Django have a local convention that avoids parens for tuples in general?

Sure I can make use `through_model` instead.